google/cloud/common_options.h

```diff
@@ -106,6 +107,111 @@ struct AuthorityOption {
   using Type = std::string;
 };
 
+/**
+ * The configuration for a HTTP Proxy.
+ *
+ * This configuration can be used for both REST-based and gRPC-based clients.
+ * The client library sets the underlying configuration parameters based on
+ * the values in this struct.
+ *
+ * The full URI is constructed as:
+ *
+ * {scheme}://{username}:{password}@{host}:port
+ *
+ * Any empty values are omitted, except for the `scheme` which defaults to
+ * `https`. If the `hostname` value is empty, no HTTP proxy is configured.
+ */
+class ProxyConfig {
+ public:
+  ProxyConfig() = default;
+
+  /// Configure the HTTP proxy host.
+  std::string const& hostname() const { return hostname_; }
+
+  /// Configure the HTTP proxy port.
+  std::string const& port() const { return port_; }
+
+  /// Configure the HTTP username.
+  std::string const& username() const { return username_; }
+
+  /// Configure the HTTP password.
+  std::string const& password() const { return password_; }
+
+  /**
+   * Set the scheme for the proxy.
+   *
+   * Use `http` or `https` the client library will append the separator (`://`)
+   * as needed. If empty, the library defaults to `https`.
+   */
+  std::string const& scheme() const { return scheme_; }
```

Comment:
"Configure" and "Set" seem inappropriate for these accessors.

Reply:
Fixed I think.